### PR TITLE
B 106 unfinish card

### PR DESCRIPTION
**Pull request by RomanKlishch**

### CODE REVIEW

File: src/main/java/com/greetingcard/service/impl/DefaultCardService.java [hunk 1]
```diff
@@ -52,9 +52,13 @@ public void deleteCardById(long cardId, long userId) {
...
 
     @Override
-    public void changeCardStatusAndCreateCardLink(Status status, long cardId) {
+    public void changeCardStatusAndCreateCardLink(String statusName, long cardId) {
+        Status status = Status.getByName(statusName);
         String hash = UUID.randomUUID().toString().replaceAll("/", "");
         cardDao.changeCardStatusAndSetCardLinkById(status, cardId, hash);
+//    public void changeCardStatus(String statusName, long cardId) {
+//        Status status = Status.getByName(statusName);
+//        cardDao.changeCardStatusById(status, cardId);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/service/impl/DefaultCardService.java:61]: Commented code

**Review by AlexKievDeveloper — DISMISSED** [review]

File: src/main/java/com/greetingcard/service/impl/DefaultCardService.java (hunk 1 above)
Comment (RomanKlishch) [reply src/main/java/com/greetingcard/service/impl/DefaultCardService.java]: Done

File: src/main/java/com/greetingcard/web/controller/CardController.java [hunk 2]
```diff
@@ -86,10 +86,10 @@
...
 
-    @PutMapping("card/{id}/status")
-    public void changeStatusAndCreateCardLink(@PathVariable long id) {
+    @PutMapping("card/{id}/status/{statusName}")
+    public void changeStatusAndCreateCardLink(@PathVariable long id, @PathVariable String statusName) {
         log.info("Received PUT request");
-        cardService.changeCardStatusAndCreateCardLink(Status.ISOVER, id);
+        cardService.changeCardStatusAndCreateCardLink(statusName, id);
         log.info("Successfully changed card status for card id: {}", id);
     }
```
Comment (Yuliia-Sh) [inline src/main/java/com/greetingcard/web/controller/CardController.java:94]: logs don't say what is changed in the system

Comment (RomanKlishch) [reply src/main/java/com/greetingcard/web/controller/CardController.java:94]: done

**Review by Yuliia-Sh — APPROVED** [review]

File: src/main/ui/src/services/cardService.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ function leaveCard(id) {
 }
 
 function finishCard(id) {
-    return serverService.sendRequest(`/card/${id}/status`, 'PUT');
+    return serverService.sendRequest(`/card/${id}/status/ISOVER`, 'PUT');
```
Comment (nomarchia) [inline src/main/ui/src/services/cardService.js:42]: Is it case sensitive? Or browser will read it as a lowercase word?

**Review by nomarchia — APPROVED** [review]

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/ui/src/services/cardService.js (hunk 3 above)
Comment (Yuliia-Sh) [reply src/main/ui/src/services/cardService.js:42]: good question, Roma promised to check that it works

**Pull request merged**